canonical-web-and-design/maas-ui
# feat(ui): build generic TooltipButton component and use in subnet details page

## Done

- Built `TooltipButton` component which sets some sensible defaults for rendering an accessible tooltip
- Used `TooltipButton` in subnet details page and subnet edit form

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to a subnet details page and check that you can tab to the tooltips in order to display them
- Open the subnet edit form and check that you can tab to the tooltips there too

## Fixes

Fixes canonical-web-and-design/app-tribe#880

## Screenshot

![Peek 2022-05-03 11-51](https://user-images.githubusercontent.com/25733845/166393975-aca6cece-3462-4358-8c6f-89fea1c31c04.gif)
